search/includes/classes/class-queue.php

```diff
@@ -1080,18 +1080,25 @@ public function get_average_queue_wait_time() {
 
 		$table_name = $this->schema->get_table_name();
 
-		$average_wait_time = $wpdb->get_var(
+		$queue_stats = $wpdb->get_row(
+			// Cannot prepare table name. @codingStandardsIgnoreStart
 			$wpdb->prepare(
-				"SELECT FLOOR( AVG( TIMESTAMPDIFF( SECOND, queued_time, NOW() ) ) ) AS average_wait_time FROM $table_name WHERE 1" // Cannot prepare table name. @codingStandardsIgnoreLine
+				"SELECT
+					FLOOR( AVG( TIMESTAMPDIFF( SECOND, queued_time, NOW() ) ) ) AS average_wait_time,
+					FLOOR( MAX( TIMESTAMPDIFF( SECOND, queued_time, NOW() ) ) ) AS longest_wait_time,
+					COUNT( * ) AS queue_count
+				FROM $table_name
+				WHERE 1"
 			)
+			// @codingStandardsIgnoreEnd
 		);
 
 		// Null value will usually mean empty table
-		if ( is_null( $average_wait_time ) || ! is_numeric( $average_wait_time ) ) {
+		if ( is_null( $queue_stats ) || ! is_object( $queue_stats ) ) {
 			return 0;
 		}
 
-		return intval( $average_wait_time );
+		return $queue_stats;
```

Comment:
I see we are no longer checking for `is_numeric`. I suggest we'd do it here and also cast the result inside the function, in that way, no error handling has to be done from the caller.

Reply:
Great point, fixed!